src/app/_hooks/useSetParams.js

```diff
@@ -0,0 +1,23 @@
+import { useRouter, useSearchParams } from 'next/navigation';
+
+export function useSetParam(param) {
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const newParams = new URLSearchParams(searchParams);
+  const addParam = value => {
+    if (searchParams.get(param) && param !== 'district') {
```

Comment:
🟠 why you dislike 'district' so much? :'(

Reply:
With districts, we have multiselect and with other filters there is only one option to be selected, so where we have one option we need to rewrite it and with districts we dont need to rewrite other districts, we need to just append new "district=value"

Reply:
that is totally valid logic, but this hook sounds and placed in a way it should be generic, that means it should not contain any logic specific to some exact param...

so i would assume you OR make hook work with both addind/replacing, OR make it super specific to your use case and place is somewhere near usage place

personally i would add some work into this hook, and make it support both 'adding' and 'replacing' values, because otherwise it looks great as it is